alphabase/peptide/smiles.py

```diff
@@ -0,0 +1,21 @@
+import functools
+
+def requires_rdkit(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            from rdkit import Chem
+        except ImportError:
+            raise ImportError(
+                f"RDKit is required for the function '{func.__name__}'. "
+                "Please install it with 'pip install alphabase[smiles]'"
+            )
+        return func(*args, **kwargs)
+    return wrapper
+
+@requires_rdkit
```

Comment:
this is not code that is actually used, right?

Reply:
yes, just as a reference for @boopthesnoot